diff --git a/node_modules/parse-path/lib/index.js b/node_modules/parse-path/lib/index.js
index 3ae4bb7..e3d383c 100644
--- a/node_modules/parse-path/lib/index.js
+++ b/node_modules/parse-path/lib/index.js
@@ -89,11 +89,12 @@ function parsePath(url) {
     splits = output.resource.split(":");
     if (splits.length === 2) {
         output.resource = splits[0];
-        if (splits[1]) {
-            output.port = Number(splits[1]);
-            if (isNaN(output.port)) {
+        const port = splits[1];
+        if (port) {
+            output.port = Number(port);
+            if (isNaN(output.port) || port.match(/^\d+$/) === null) {
                 output.port = null;
-                parts.unshift(splits[1]);
+                parts.unshift(port);
             }
         } else {
             output.port = null;
